### PR TITLE
Don't advance the wallet init timer until we're done reindexing.

### PR DESCRIPTION
**Pull request by nuttycom**

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -625,8 +625,15 @@ void ThreadStartWalletNotifier()
         if (walletBestBlockHash.has_value()) {
             int64_t slept;
             auto timedOut = [&]() -> bool {
-                MilliSleep(50);
-                slept += 50;
+                MilliSleep(500);
+                {
+                    LOCK(cs_main);
+                    bool isReindexing{false};
+                    if (pblocktree->ReadReindexing(isReindexing) && !isReindexing) {
```
Comment (ebfull) [inline src/init.cpp:632]: (not blocking) This causes a db read; perhaps once `isReindexing` is `false` once we can stop calling `ReadReindexing`.

File: src/init.cpp [hunk 2]
```diff
@@ -625,8 +625,15 @@ void ThreadStartWalletNotifier()
         if (walletBestBlockHash.has_value()) {
             int64_t slept;
```
[the remark this answers is not in the document]
Comment (nuttycom) [reply src/init.cpp:626]: This is the one that's fixed in the previous PR, I just need to rebase.

**Review by daira — APPROVED** [review]
utACK

**Review by ebfull — APPROVED** [review]

**Review by ebfull — APPROVED** [review]

File: src/init.cpp [hunk 3]
```diff
@@ -624,9 +624,23 @@ void ThreadStartWalletNotifier()
 
         if (walletBestBlockHash.has_value()) {
             int64_t slept{0};
+            bool isReindexing{true};
             auto timedOut = [&]() -> bool {
-                MilliSleep(50);
-                slept += 50;
+                MilliSleep(500);
+                {
+                    LOCK(cs_main);
```
Comment (daira) [inline src/init.cpp:631]:
```suggestion
```
This lock is unnecessary.

File: src/init.cpp [hunk 4]
```diff
@@ -624,9 +624,22 @@ void ThreadStartWalletNotifier()
...
+                    if (isReindexing) {
+                        LOCK(cs_main);
+                        pblocktree->ReadReindexing(isReindexing);
+                    }
+
+                    // once we're out of reindexing, we can start incrementing the slept counter
+                    if (!isReindexing) {
+                        slept += 500;
+                    }
+                }
```
Comment (daira) [inline src/init.cpp:641]: This could be unindented.

Comment (daira) [reply src/init.cpp]: Yeah, I was going to edit my earlier suggestion to note that, but you were too fast for me :-)

File: src/init.cpp (hunk 3 above)
Comment (nuttycom) [reply src/init.cpp:632]: `blocktree` doesn't need to be guarded by a lock?

**Review by daira — APPROVED** [review]
utACK

File: src/init.cpp [hunk 5]
```diff
@@ -624,9 +624,20 @@ void ThreadStartWalletNotifier()
...
             int64_t slept{0};
+            bool isReindexing{true};
             auto timedOut = [&]() -> bool {
-                MilliSleep(50);
-                slept += 50;
+                MilliSleep(500);
+                // re-check whether we're reindexing
+                if (isReindexing) {
+                    LOCK(cs_main);
+                    pblocktree->ReadReindexing(isReindexing);
```
Comment (LarryRuane) [inline src/init.cpp:633]: Could this just test `fReindex` here? It should be set by the time we reach this code (whether `-reindex` is specified in the config or read from the index). It's an atomic boolean so wouldn't need `cs_lock`. Also, you wouldn't need the local `isReindexing` variable. But this should work.

**Review by LarryRuane — COMMENTED** [review]
Post-merge ACK, one suggestion but optional.

**Pull request merged**